### DISCUSSION TITLE
XmlReader/XmlWriter plans

### DISCUSSION THREAD

**Discussion by lechu445** [dotnet/runtime#52809] [Q&A] [answered]
As still many enterprises does use communication over XML (e.g. SOAP) does the team has any plans to enhance XML processing in terms of performance?

- is there any plan to implement different library in similar way to `System.Text.Json`?
- are there any plans to extend XmlReader/XmlWriter public API to handle `Span<T>` or `Memory<T>` inputs?
- if none of these improvements are planned, what are suggestions to use as an alternative?

**Comment by lechu445** [comment]
@jeffhandley

**Comment by jeffhandley** [comment, marked as answer]
Hi @lechu445. I'm sorry for the terribly long delay responding.

We do not have any plans to proactively enhance XML processing performance in the ways you've described. If you have scenarios where the `Span<T>` or `Memory<T>` inputs would be valuable, I recommend filing an issue with those scenarios described. Once the scenario is understood and we're able to determine the potential benefits, we could consider an API proposal to add `Span<T>`/`Memory<T>` support.

Thanks for the question and your patience.
